cunyfirstapi/cunyfirstapi.py

```diff
@@ -91,12 +91,113 @@ def to_login(self):
+        url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
+        response = self._session.get(url)
+        #print(r.text)
+        tree = html.fromstring(response.text)                  # parse the html DOM
+        data = {}                                       # store the key-value pairs for the form here
+        for el in tree.xpath('//input'):
+            # iterate through the hidden form
+            name = ''.join(el.xpath('./@name'))
+            value = ''.join(el.xpath('./@value'))
+            #print(name,value)
+            data[name] = value
+        
+        # manually make some changes to the form
+        data['ICAJAX'] = '1'
+        data['ICNAVTYPEDROPDOWN'] = '1'
+        data['ICYPos'] = '144'
+        data['ICStateNum'] = '1'
+        data['ICAction'] = 'DERIVED_SSS_SCR_SSS_LINK_ANCHOR4'
+        data['ICBcDomData'] = ''
+        data['DERIVED_SSS_SCL_SSS_MORE_ACADEMICS'] = '9999'
+        data['DERIVED_SSS_SCL_SSS_MORE_FINANCES'] = '9999'
+        data['CU_SF_SS_INS_WK_BUSINESS_UNIT'] = college_code
+        data['DERIVED_SSS_SCL_SSS_MORE_PROFILE'] = '9999'
+
+        # set url to student center menu
+        url = constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL
+        response = self._session.post(url = url, data = data)                 # post data to student center menu
+       
+        # navigate to the academics page
+        self._session.get(constants.CUNY_FIRST_MY_ACADEMICS_URL)
+        
+        # modify form for next stage
+        data['ICStateNum'] = '3'
+        data['ICAction'] = 'DERIVED_SSSACA2_SS_UNOFF_TRSC_LINK'
+        data['ICYPos'] = '95'
+        data['DERIVED_SSTSNAV_SSTS_MAIN_GOTO$7$'] = '9999'
+        data['DERIVED_SSTSNAV_SSTS_MAIN_GOTO$8$'] = '9999'
+
+        # go to transcript request page by posting data saying we want to go
+        response = self._session.post(url, data = data)
+
+        #data['url'] = url
+
+        url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
+
+        response = self._session.get(url)            # actually go to transcript request page
+
+        return response
+
+    def download_transcript(self, college_code, data = None):
+        if data is None:
+            data = {'ICElementNum': '0'}
+        # modify the form data to say we declared a college to pick from
+
+        url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
+        r = self._session.get(url)
+        #print(re.search(r'ICSID.*',r.text))
+        tree = html.fromstring(r.text)
+        data['ICAJAX'] = '1'
+        data['ICSID'] = tree.xpath('//*[@id="ICSID"]/@value')[0]
+
+        data['ICStateNum'] = '5'
+        data['ICAction'] = 'SA_REQUEST_HDR_INSTITUTION'
+        data['SA_REQUEST_HDR_INSTITUTION'] = college_code
+        data['ICYPos'] ='115'
+        data['ICBcDomData'] = ''
+        data['DERIVED_SSTSRPT_TSCRPT_TYPE3'] = ''
+        data['ptus_componenturl'] = 'https://hrsa.cunyfirst.cuny.edu/psp/cnyhcprd/EMPLOYEE/HRMS/c/SA_LEARNER_SERVICES.SSS_TSRQST_UNOFF.GBL'
```

Comment:
why isnt this URL in the constants?

Reply:
We may not even need it in this code, I will look into removing it